src/sentry/integrations/opsgenie/client.py

```diff
@@ -47,9 +51,54 @@ def metadata(self):
 
     # This doesn't work if the team name is "." or "..", which Opsgenie allows for some reason
     # despite their API not working with these names.
-    def get_team_id(self, integration_key: str, team_name: str) -> BaseApiResponseX:
+    def get_team_id(self, team_name: str) -> BaseApiResponseX:
         params = {"identifierType": "name"}
         quoted_name = quote(team_name)
         path = f"/teams/{quoted_name}"
-        headers = {"Authorization": "GenieKey " + integration_key}
+        headers = {"Authorization": "GenieKey " + self.integration_key}
         return self.get(path=path, headers=headers, params=params)
+
+    def send_notification(self, data, rules):
+        if isinstance(data, (Event, GroupEvent)):
+            group = data.group
+            event = data
+            if group is None:
+                payload = {
+                    "message": event.message or event.title,
+                    "source": "Sentry",
+                    "details": {
+                        "Triggering Rules": ", ".join([rule.label for rule in rules]),
+                    },
+                    "tags": [
+                        f'{str(x).replace(",", "")}:{str(y).replace(",", "")}'
+                        for x, y in event.tags
+                    ],
+                }
```

Comment:
All of the fields that appear in this payload are shared with the payload below. So we can instead instantiate it like this every time, and modify it if the group is present:

```python
payload = {} # message, source, details, tags
if group:
  payload["alias"] = ...
  payload["details"]["Sentry ID"] = ...
  payload["entity"] = ...
  # etc
```